src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -0,0 +1,228 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+import os
+import queue
+import traceback
+from enum import Enum
+from typing import Any, Tuple, Optional, Callable, Dict, Set
+
+from snowflake.snowpark._internal.data_source.dbms_dialects.sqlite3_dialect import (
+    Sqlite3Dialect,
+)
+from snowflake.snowpark._internal.data_source.drivers.sqlite_driver import SqliteDriver
+import snowflake
+from snowflake.snowpark._internal.data_source.datasource_reader import DataSourceReader
+from snowflake.snowpark._internal.data_source.dbms_dialects.oracledb_dialect import (
+    OracledbDialect,
+)
+from snowflake.snowpark._internal.data_source.dbms_dialects.sqlserver_dialect import (
+    SqlServerDialect,
+)
+from snowflake.snowpark._internal.data_source.drivers.oracledb_driver import (
+    OracledbDriver,
+)
+from snowflake.snowpark._internal.data_source.drivers.pyodbc_driver import PyodbcDriver
```

Comment:
I recommend to update dbms_dialect/__init__.py and drivers/__init__.py to have `ALL = [...]`
so that the import can be much easier

```python
from snowflake.snowpark._internal.data_source.drivers import 
(
    SqliteDriver,
    OracledbDriver,
    PyodbcDriver
)

... same to the dialects
```